### PR TITLE
Updated Repo interface

### PR DESCRIPTION
**Pull request by g4s8**
#57 - change signature of `Repo` interface to use `ByteBuffer` for manifest reading.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#58](https://github.com/artipie/docker-adapter/issues/58) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/ae3690f7-364b-4bcb-9581-1b669942d2fa); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/docker/Repo.java [hunk 1]
```diff
@@ -52,5 +52,5 @@
      * @param link Manifest reference link
      * @return Future with manifest JSON object
```
Comment (fanifieiev) [inline src/main/java/com/artipie/docker/Repo.java]: @g4s8 Please remove JSON word.

**Comment by g4s8** [comment]
@fanifieiev done, fixed: https://github.com/artipie/docker-adapter/pull/58/commits/ec08c5f80276597d2c24c168178ce599021da148

**Comment by fanifieiev** [comment]
@g4s8 Good to merge

**Comment by 0crat** [comment]
Job was finished in 44 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: performer @fanifieiev[/z](https://www.0crat.com/u/fanifieiev)[/z](https://www.0crat.com/u/fanifieiev) didn't make CR comments

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**